lib/components/primitive-components/Group/Group.ts

```diff
@@ -152,7 +152,7 @@ export class Group<Props extends z.ZodType<any, any, any> = typeof groupProps>
           {
             method: "POST",
             body: JSON.stringify({
-              input_circuit_json: this.root!.db.toArray(),
+              input_circuit_json: this.root!.getCircuitJson(),
             }),
```

Comment:
GetCircuitJson should only be used by users, not internally (because it can cause rerenders, but db.toArray cant)

Reply:
But db.toArray has the default width and height, and not the processed width and height after the calculation

Reply:
That’s a render phase issue then, we still should not call getCircuitJson. We can eg approximate the board size in a render phase prior to autorouting